File: app/src/main/java/de/chrisward/theyworkforyou/repository/LordRepository.java
package de.chrisward.theyworkforyou.repository;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.content.Context;

import java.util.List;

import de.chrisward.theyworkforyou.R;
import de.chrisward.theyworkforyou.api.LordApi;
import de.chrisward.theyworkforyou.model.Lord;
import de.chrisward.theyworkforyou.wrapper.Resource;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LordRepository extends BaseRepository {
    private static LordRepository instance;
    private LordApi lordApi;

    private LordRepository(final String apiKey) {
        super(apiKey);
        lordApi = getRetrofit().create(LordApi.class);
    }

    public synchronized static LordRepository getInstance(Context context) {
        if (instance == null) {
            instance = new LordRepository(
                    context.getResources().getString(R.string.theyworkforyou_api_key)
            );
        }

        return instance;
    }

    public LiveData<Resource<List<Lord>>> refreshLords() {
        final MutableLiveData<Resource<List<Lord>>> data = new MutableLiveData<>();

        lordApi.getMPs().enqueue(new Callback<List<Lord>>() {
            @Override
            public void onResponse(Call<List<Lord>> call, Response<List<Lord>> response) {
                if (response.body() == null) {
                    return;
                }
                data.setValue(new Resource<>(Resource.SUCCESS, response.body(), null));
            }

            @Override
            public void onFailure(Call<List<Lord>> call, Throwable t) {
                data.setValue(new Resource<>(Resource.ERROR, null, t.getMessage()));
            }
        });

        return data;
    }
}

File: app/src/main/java/de/chrisward/theyworkforyou/repository/LordRepository.kt
package de.chrisward.theyworkforyou.repository

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import de.chrisward.theyworkforyou.api.LordApi
import de.chrisward.theyworkforyou.model.Lord
import de.chrisward.theyworkforyou.wrapper.Resource
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import javax.inject.Inject

class LordRepository @Inject constructor(private val lordApi: LordApi) {
    fun refreshLords(): LiveData<Resource<List<Lord>>> {
        val data = MutableLiveData<Resource<List<Lord>>>()

        lordApi.mPs.enqueue(object : Callback<List<Lord>> {
            override fun onResponse(call: Call<List<Lord>>, response: Response<List<Lord>>) {
                if (response.body() == null) {
                    return
                }
                data.value = Resource(Resource.Status.SUCCESS, response.body())
            }

            override fun onFailure(call: Call<List<Lord>>, t: Throwable) {
                data.value = Resource(status = Resource.Status.ERROR, message = t.message)
            }
        })

        return data
    }
}
